### PR TITLE
georgettica/bump slowly to latest operatorsdk [OSD-6289]

### PR DESCRIPTION
**Pull request by georgettica**
currently final verison is 1.13.0

### CODE REVIEW

**Comment by georgettica** [comment]
waiting on https://github.com/openshift/release/pull/18918
as it will make the prow things pass

**Comment by georgettica** [comment]
/retest
To see if the new CI works

**Comment by georgettica** [comment]
Seems there are build issue..

Will fix and unhold again

**Comment by georgettica** [comment]
🤞

**Comment by georgettica** [comment]
`/retest`
seems I cannot /draft it while tests are running
I guess
/hold
is in place

**Comment by georgettica** [comment]
I am working on a local branch to get this to work

So far it's going well and soon I will submit a PR for it

**Comment by 2uasimojo** [comment]
Note that you *should* be able to `container-make vendor` from any platform and get the right result.

**Comment by georgettica** [comment]
I can test that, and this should work

**Comment by georgettica** [comment]
I want to add some more logging to the hack/teat-integration in order to make  it more readable

**Comment by georgettica** [comment]
partial PR
https://github.com/openshift/route-monitor-operator/pull/137

**Comment by georgettica** [comment]
Seems the code complains on the namespace not existing..
Ideas?

**Comment by georgettica** [comment]
new operator-sdk was introduced, and migration doc shows nothing to be done for our operator

**Comment by georgettica** [comment]
The current issue is the controller-gen is not updating on the ci server

**Comment by 2uasimojo** [comment]
> The current issue is the controller-gen is not updating on the ci server

Should it be? According to the [controller-gen target](https://github.com/openshift/route-monitor-operator/pull/120/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R106), it'll only install the binary if there isn't already one in $PATH. But there is, in the [backing image](https://github.com/openshift/boilerplate/blob/328a6a5330ca758439829122dbdfafca9dcc79cf/config/build_image-v2.0.0.sh#L45) CI uses.

If you want this to pass as is, you need to generate your CRDs using that same version (which you could do by running `make generate` from within `quay.io/app-sre/boilerplate:image-v2.1.0`); or change the logic in the `controller-gen` target to override the install if it finds an existing binary at the wrong version.

**Comment by georgettica** [comment]
If by any chance @2uasimojo do you know how to do 
```
If
Else if
Else
```
In makefile that would be great ❤️

**Comment by 2uasimojo** [comment]
https://www.gnu.org/software/make/manual/html_node/Conditional-Syntax.html

**Comment by georgettica** [comment]
@2uasimojo seems this was the issue, a space :( https://stackoverflow.com/questions/34761685/extraneous-text-after-else-directive

**Comment by georgettica** [comment]
Fixed that issue by saving the binary locally and added an issue to have a docker image option to it 
Ready for review

File: controllers/clusterurlmonitor/clusterurlmonitor.go
Comment (NautiluX) [inline controllers/clusterurlmonitor/clusterurlmonitor.go:69]: Why do we need a different key?

File: controllers/clusterurlmonitor/clusterurlmonitor_supplement.go
Comment (NautiluX) [inline controllers/clusterurlmonitor/clusterurlmonitor_supplement.go:114]: What if we fail to delete the prevfinalizer?

File: controllers/clusterurlmonitor/clusterurlmonitor_supplement.go
Comment (NautiluX) [inline controllers/clusterurlmonitor/clusterurlmonitor_supplement.go:124]: What if we fail to delete the prevfinalizer?

File: controllers/suite_test.go
Comment (NautiluX) [inline controllers/suite_test.go:79]: Can we remove this code if we don't need it?

File: main.go
Comment (NautiluX) [inline main.go:75]: Is this related to the version upgrade? If not can we separate it into a different PR?

File: main.go
Comment (NautiluX) [inline main.go:65]: Is this related to the version upgrade? If not can we separate it into a different PR?

File: pkg/util/finalizer/patch_finalizer.go
Comment (NautiluX) [inline pkg/util/finalizer/patch_finalizer.go:7]: Is this function used anywhere?

**Review by NautiluX — CHANGES_REQUESTED** [review]
It looks like this PR includes a lot of changes, not all of them seem to be related to the version upgrade. Please at least rebase before merge to structure changes so we don't merge this looong list of commits. It looks like they rather track your work than explain what the changes do.
/approve cancel

File: controllers/clusterurlmonitor/clusterurlmonitor.go
Comment (georgettica) [reply controllers/clusterurlmonitor/clusterurlmonitor.go:69]: This is a part of the migration, one of the commits talks about it

File: controllers/suite_test.go
Comment (georgettica) [reply controllers/suite_test.go:79]: It's from operator-sdk, which means it's gonna come back of we want it?

File: main.go
Comment (georgettica) [reply main.go:75]: it is related, we are using the config with `config/manager/controller_manager_config.yaml` and to load it we need this new var

File: main.go
Comment (georgettica) [reply main.go:65]: this was the reason we did the version upgrade, to have the liveness and readiness

File: pkg/util/finalizer/patch_finalizer.go
Comment (georgettica) [reply pkg/util/finalizer/patch_finalizer.go:7]: nope, it was supposed to and got dismissed (sad because I have tests for it too)

File: controllers/clusterurlmonitor/clusterurlmonitor_supplement.go
Comment (georgettica) [reply controllers/clusterurlmonitor/clusterurlmonitor_supplement.go:124]: the `DeleteFinalizer` is a local command only, so it just removes from an array if the item exists or keeps it empty

also the `s.Common.UpdateMonitorResource` returns the error if something goes sideways

File: controllers/clusterurlmonitor/clusterurlmonitor_supplement.go
Comment (georgettica) [reply controllers/clusterurlmonitor/clusterurlmonitor_supplement.go:114]: same

**Review by NautiluX — COMMENTED** [review]
/lgtm
/hold
unhold to merge when ready

**Review by NautiluX — APPROVED** [review]

**Comment by georgettica** [comment]
@NautiluX so it's ready to unhold?

**Pull request merged**
